_commands/tcpdump.md

```diff
@@ -0,0 +1,75 @@
+22:59:52.031854 IP bud02s22-in-f196.1e100.net.http > 10.0.2.15.56854: Flags [P.], seq 109760840:109761678, ack 1151493304, win 65535, length 838: HTTP: HTTP/1.1 302 Found
+22:59:52.031870 IP 10.0.2.15.56854 > bud02s22-in-f196.1e100.net.http: Flags [.], ack 109761678, win 31844, length 0
+22:59:52.035854 IP 10.0.2.15.59170 > bud02s22-in-f196.1e100.net.https: Flags [P.], seq 2068573025:2068573116, ack 110036572, win 65320, length 91
+22:59:52.036112 IP bud02s22-in-f196.1e100.net.https > 10.0.2.15.59170: Flags [.], ack 2068573116, win 65535, length 0
+22:59:52.277901 IP bud02s22-in-f196.1e100.net.https > 10.0.2.15.59170: Flags [P.], seq 110036572:110036648, ack 2068573116, win 65535, length 76
+22:59:52.280812 IP bud02s22-in-f196.1e100.net.https > 10.0.2.15.59170: Flags [P.], seq 110036648:110040938, ack 2068573116, win 65535, length 4290
+22:59:52.280833 IP 10.0.2.15.59170 > bud02s22-in-f196.1e100.net.https: Flags [.], ack 110040938, win 65535, length 0
+...
+... (continues)
+~~~
+##### Break it down
+-The `-S` flag prints absolute TCP sequence numbers versus relative
+-host <hostname> lets you specify a specific hostname to track requests and responses from a specific host (i.e. Google, Yahoo, eecs.umich.edu, etc.)
+#### Example command
+
+`tcpdump -S "tcp[tcpflags] & (tcp-syn|tcp-ack|tcp-fin) != 0"`
+
+~~~ bash
+$ tcpdump -S "tcp[tcpflags] & (tcp-syn|tcp-ack|tcp-fin) != 0"
+tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
+listening on enp0s3, link-type EN10MB (Ethernet), capture size 262144 bytes
+23:13:35.006934 IP 10.0.2.15.44258 > ec2-204-236-230-155.compute-1.amazonaws.com.https: Flags [.], ack 210881678, win 32660, length 0
+23:13:35.007386 IP ec2-204-236-230-155.compute-1.amazonaws.com.https > 10.0.2.15.44258: Flags [.], ack 2435184464, win 65535, length 0
+23:13:38.226091 IP live.github.com.https > 10.0.2.15.47844: Flags [P.], seq 75075962:75075993, ack 2779464789, win 65535, length 31
+23:13:38.226285 IP 10.0.2.15.47844 > live.github.com.https: Flags [P.], seq 2779464789:2779464824, ack 75075993, win 39760, length 35
+23:13:38.226541 IP live.github.com.https > 10.0.2.15.47844: Flags [.], ack 2779464824, win 65535, length 0
+23:13:45.022884 IP 10.0.2.15.44258 > ec2-204-236-230-155.compute-1.amazonaws.com.https: Flags [.], ack 210881678, win 32660, length 0
+23:13:45.023070 IP ec2-204-236-230-155.compute-1.amazonaws.com.https > 10.0.2.15.44258: Flags [.], ack 2435184464, win 65535, length 0
+...
+... (continues)
+~~~
+##### Break it down
+- This command will let you watch the 3-way handshake between different clients and servers that are functioning on your network. Specifies that tcpflags are active and that at least one of the syn, ack, or fin flags are selected.
+- You can save this output or the above output with the redirection >.
+- NOTE: you can obviously combine the above two and specify a specific host to watch the 3-way handshake on for more specified traffic monitoring.
+- NOTE: 
```

Comment:
Oops. Must not have saved. Will update.
On Mar 25, 2016 11:32 PM, "Matt Terwilliger" <notifications@github.com>
wrote:

> In _commands/tcpdump.md
> <https://github.com/c4cs/c4cs.github.io/pull/183#discussion_r57508422>:
>
> > +listening on enp0s3, link-type EN10MB (Ethernet), capture size 262144 bytes
> > +23:13:35.006934 IP 10.0.2.15.44258 > ec2-204-236-230-155.compute-1.amazonaws.com.https: Flags [.], ack 210881678, win 32660, length 0
> > +23:13:35.007386 IP ec2-204-236-230-155.compute-1.amazonaws.com.https > 10.0.2.15.44258: Flags [.], ack 2435184464, win 65535, length 0
> > +23:13:38.226091 IP live.github.com.https > 10.0.2.15.47844: Flags [P.], seq 75075962:75075993, ack 2779464789, win 65535, length 31
> > +23:13:38.226285 IP 10.0.2.15.47844 > live.github.com.https: Flags [P.], seq 2779464789:2779464824, ack 75075993, win 39760, length 35
> > +23:13:38.226541 IP live.github.com.https > 10.0.2.15.47844: Flags [.], ack 2779464824, win 65535, length 0
> > +23:13:45.022884 IP 10.0.2.15.44258 > ec2-204-236-230-155.compute-1.amazonaws.com.https: Flags [.], ack 210881678, win 32660, length 0
> > +23:13:45.023070 IP ec2-204-236-230-155.compute-1.amazonaws.com.https > 10.0.2.15.44258: Flags [.], ack 2435184464, win 65535, length 0
> > +...
> > +... (continues)
> > +~~~
> > +##### Break it down
> > +- This command will let you watch the 3-way handshake between different clients and servers that are functioning on your network. Specifies that tcpflags are active and that at least one of the syn, ack, or fin flags are selected.
> > +- You can save this output or the above output with the redirection >.
> > +- NOTE: you can obviously combine the above two and specify a specific host to watch the 3-way handshake on for more specified traffic monitoring.
> > +- NOTE:
>
> What's the NOTE?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/c4cs/c4cs.github.io/pull/183/files/95245ea5c04f7e7e6e44ab2ec83e827a8e4a5ea0#r57508422>
>